cases/models.py

```diff
@@ -177,43 +177,43 @@ def get_merged_into_cases(self, cases):
         query = self.raw(
             """WITH RECURSIVE cte AS (
                  SELECT
-                   c.id,
-                   c.merged_into_id,
-                   last_merge.time
+                   m.mergee_id AS id,
+                   m.merged_into_id,
+                   m.time
                  FROM
-                   cases_case c
-                   JOIN (
-                     SELECT m.mergee_id, m.merged_into_id, MAX(m.time) AS time
-                     FROM cases_mergerecord m
-                     WHERE m.unmerge IS NOT TRUE
-                     GROUP BY m.mergee_id, m.merged_into_id
-                   ) AS last_merge ON last_merge.merged_into_id = c.merged_into_id
-                   AND last_merge.mergee_id = c.id
+                   cases_mergerecord m
                  WHERE
-                   c.id IN (%s)
-                   AND c.merged_into_id IS NOT NULL
+                   NOT m.unmerge
+                   AND (
+                     SELECT count(*)
+                     FROM cases_mergerecord m2
+                     WHERE m2.id > m.id
+                     AND m2.mergee_id = m.mergee_id
+                     AND m2.merged_into_id = m.merged_into_id
+                    ) = 0 -- There are no later records.
+                   AND m.mergee_id IN (%s)
                  UNION
                  SELECT
-                   cte.id,
-                   c.merged_into_id,
-                   last_merge.time
+                   cte.id AS id,
+                   m.merged_into_id,
+                   m.time
                  FROM
-                   cases_case c
-                   JOIN cte ON c.id = cte.merged_into_id
-                   JOIN (
-                     SELECT m.mergee_id, m.merged_into_id, MAX(m.time) AS time
-                     FROM cases_mergerecord m
-                     WHERE m.unmerge IS NOT TRUE
-                     GROUP BY m.mergee_id, m.merged_into_id
-                   ) AS last_merge ON last_merge.merged_into_id = cte.merged_into_id
-                   AND last_merge.mergee_id = cte.id
+                   cases_mergerecord m
+                   JOIN cte ON m.mergee_id = cte.merged_into_id
                  WHERE
-                   c.merged_into_id IS NOT NULL
+                   NOT m.unmerge
+                   AND (
+                     SELECT count(*)
```

Comment:
count() is sadly definitely not performant on postgresql - perhaps adding a limit to my suggestion would work to deal with the multiple error you were presumably getting - ie. 
`(SELECT id FROM cases_mergerecord m2 WHERE m2.id > m.id AND m2.mergee_id = m.mergee_id AND m2.merged_into_id = m.merged_into_id LIMIT 1) IS NULL`